### DISCUSSION TITLE
Confusion of parameters

### DISCUSSION THREAD

**Discussion by ndvbd** [ruby-amqp/bunny#637] [General]
I see in different places different parameters, which is very strange. Not sure which ones are valid or not.

In Bunny.new I see in some places `heartbeat`, sometimes I see `heartbeat_interval`, sometimes I see `heartbeat_timeout` and `automatic_recovery`.
Is there one place showing all parameters?

**Comment by michaelklishin** [comment]
Just use what the docs mention.

Some names were picked by the original author and later retained for backwards compatibility, even when a more specific name (such as `heartbeat_interval`) was introduced.

**Comment by michaelklishin** [comment]
Bunny is a 13 year old codebase, so some naming changes and refinements were inevitable. We never dropped support for the original names, which I don't see as a major problem. Maybe one day we will.

**Comment by ndvbd** [comment]
@michaelklishin Thanks. Where is the right place to look in the docs? Looking here: [](http://reference.rubybunny.info/Bunny.html#new-class_method)
On Bunny.new , it doesnt say what are the valid values for `connection_string_or_opts`.

**Reply by michaelklishin** [reply]
All or nearly all options are listed in the `Bunny::Session#initialize` doc reference. I've ported them to `Bunny.new` and pushed. Thanks for the idea!

**Comment by ndvbd** [comment]
Thanks, but I still don't see it here: `https://www.rubydoc.info/github/ruby-amqp/bunny/Bunny#new-class_method`
Maybe I am missing something?

**Reply by michaelklishin** [reply]
I cannot speak for rubydoc.info. Maybe they only update when a new tag is detected. Bunny is open source software, so you can trivially generate docs with YARD locally.

**Comment by ndvbd** [comment]
What is the link to the Bunny docs you are referring to?

**Reply by michaelklishin** [reply]
The guides have been [moved to this repo](https://github.com/ruby-amqp/bunny/tree/main/docs/guides) a while ago.
